File: backend/app/schemas/catalog.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class CategoryBase(BaseModel):
    name: str
    slug: str
    description: Optional[str] = None
    image_url: Optional[str] = None
    parent_id: Optional[int] = None
    status: bool = True

class CategoryCreate(CategoryBase):
    pass

class Category(CategoryBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class ProductBase(BaseModel):
    sku: str
    name: str
    slug: str
    description: Optional[str] = None
    short_description: Optional[str] = None
    price: float
    special_price: Optional[float] = None
    inventory: int = 0
    status: bool = True
    image_url: Optional[str] = None
    category_id: Optional[int] = None

class ProductCreate(ProductBase):
    pass

class Product(ProductBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: app/src/main/java/com/fastshopping/models/Catalog.kt
package com.fastshopping.models

data class Product(
    val id: Int,
    val sku: String,
    val name: String,
    val slug: String,
    val description: String?,
    val price: Double,
    val inventory: Int,
    val image_url: String?,
    val category_id: Int?
)

data class Category(
    val id: Int,
    val name: String,
    val slug: String,
    val description: String?,
    val image_url: String?
)
